src/muz/spacer/spacer_global_generalizer.cpp

```diff
@@ -422,39 +432,78 @@ bool lemma_global_generalizer::skolemize_sel_vars(expr_ref &f,
     return true;
 }
 
-// compute a lemma that subsumes lemmas in lc
-bool lemma_global_generalizer::subsume(lemma_cluster lc, lemma_ref &lemma,
-                                       expr_ref_vector &subs_gen) {
-    const expr_ref &pattern = lc.get_pattern();
-    unsigned n_vars = get_num_vars(pattern);
-    SASSERT(n_vars > 0);
-    reset(n_vars);
+/// If possible, find a model for (a /\ b). If not, find model for a
```

Comment:
try phrasing this in terms of hard and soft constraints. `a` is hard`, `b` is soft, you want `maxsat` of `a` and `b`.
So maybe call this, `max_sat_with_model`. Should be allowed to return a bool so it can fail in case `a` is unsat.
Easier than commenting all the assumptions.